test/ct_test.c

```diff
@@ -497,6 +497,51 @@ static int test_ctlog_from_base64(void)
         return 0;
     return 1;
 }
+
+static int test_ctlog_store_from_ctlog(void)
+{
+    /* Google test log */
+    const uint8_t log_id[32]  = {
+        0xDF, 0x1C, 0x2E, 0xC1, 0x15, 0x00, 0x94, 0x52,
+        0x47, 0xA9, 0x61, 0x68, 0x32, 0x5D, 0xDC, 0x5C,
+        0x79, 0x59, 0xE8, 0xF7, 0xC6, 0xD3, 0x88, 0xFC,
+        0x00, 0x2E, 0x0B, 0xBD, 0x3F, 0x74, 0xD7, 0x64
+    };
+    const char *log_name;
+    const CTLOG *log;
+    EVP_PKEY *log_public_key;
+    CTLOG *log_copy;
+    CTLOG_STORE *store;
+
+    SETUP_CT_TEST_FIXTURE();
+    
+    /* Extract log from file-based store. */
+    if (!TEST_ptr(log = CTLOG_STORE_get0_log_by_id(fixture->ctlog_store,
+                                                   log_id, sizeof(log_id))))
+        return 0;
+    log_name = CTLOG_get0_name(log);
+    log_public_key = CTLOG_get0_public_key(log);
+    if (!TEST_true(EVP_PKEY_up_ref(log_public_key)))
+        return 0;
+    if (!TEST_ptr(log_copy = CTLOG_new(log_public_key, log_name)))
+        return 0;
```

Comment:
Leaks `log` and `log_public_key` on failure. More similar leaks below. Probably having a common cleanup `err` block to jump to via `goto` would be the way to go.

Reply:
I recommend running valgrind locally when adding a new test..

Reply:
Done.